plutus-benchmark/marlowe/exe/Main.hs

```diff
@@ -46,39 +46,39 @@ main =
     benchmarks <- either error id <$> Semantics.benchmarks
 
     -- Write the tabulation of semantics benchmark results.
-    writeFile "marlowe-semantics.tsv"
+    writeFile "marlowe/exe/marlowe-semantics.tsv"
```

Comment:
This seems to be where the program fails if you run `marlowe-semantics` in the wrong place.  It seems that if you call `writeFile` with a path like `dir/file.txt` and `dir` doesn't exist then it fails kind of unhelpfully with `*** Exception: dir/file.txt : withFile: does not exist (No such file or directory)`.  If the directory does exist then you're OK, so if I run `marlowe-semantics` in a directory with an empty `marlowe/exe` then everything works.  I think that there may be three ways to fix this: (1) get it to check that `marlowe/exe` exists first; (2) use `getDataDir` like it does later in this file to make sure that it's writing the results into the correct directory; (3) just write the files in the current directory, where the user started the program.  I'm not sure what's best (but probably not 1).  It finds the benchmark files and the flat files OK no matter where you run it (as long as you're inside `plutus`), so I think this (and some similar things in this file) is the only thing that's forcing you to run it in `plutus-benchmark`.

Reply:
thanks. I did (2).

Reply:
Yes, that all seems to work now.  I can run it anywhere without any problems.